addons/mrp/tests/test_warehouse_multistep_manufacturing.py

```diff
@@ -810,3 +810,28 @@ def test_manufacturing_bom_with_repetitions(self):
         mo_P2 = self.env['mrp.production'].search([('product_id', '=', products[2].id)])
         self.assertEqual(mo_P1.product_uom_qty, 5.0)
         self.assertEqual(mo_P2.product_uom_qty, 5.0)
+
+    def test_update_component_qty(self):
+        warehouse = self.warehouse
+        warehouse.manufacture_steps = "pbm"
+        bom = self.bom
+        product = bom.product_id
+        component = bom.bom_line_ids.product_id
+        mo = self.env['mrp.production'].create({
+            'product_id': product.id,
+            'bom_id': bom.id,
+            'product_qty': 1,
+            'location_src_id': warehouse.pbm_loc_id.id,
+        })
+        mo.action_confirm()
+        self.assertEqual(mo.move_raw_ids.product_qty, 2.0)
```

Comment:
```suggestion
        self.assertEqual(mo.move_raw_ids.product_uom_qty, 2.0)
```
Doesn't change much in this case, but let's avoid a potential mess with the `uom`. Same remark for the others `move.product_qty` after this line.